app/carbonapi/http_handlers.go

```diff
@@ -214,9 +215,23 @@ func (app *App) renderHandler(w http.ResponseWriter, r *http.Request) {
 		targetErr, metricSize := app.getTargetData(targetCtx, target, exp, metricMap,
 			&results, &form, &toLog, logger, &partiallyFailed)
 		if targetErr != nil {
+			// we can have 3 error types here
+			// a) dataTypes.ErrNotFound  > Continue, at the end we check if all errors are 'not found' and we answer with http 404
+			// b) parser.ParseError -> Return with this error(like above, but with less details )
+			// c) anything else -> continue, answer will be 5xx if all targets have one error
```

Comment:
I think returning with 5xx makes more sense.

Reply:
Yes, agree! And then we will not optimistFanIn for target errors. 

I just want to implement this change in smaller steps so we could see better the error types.
After this change, we will see better what type are the remaining errors and what type are they supposed to be. The reason I did not implement it now was that I'm not sure if some of the errors from c) can be 'ErrNotFound' masked as 'generic error' or not.

The plan is to roll this and then we can have better view on remaining errors.